### ISSUE TITLE
Not compatible with history v5?

### ISSUE THREAD

**Issue by Guymestef** [supasate/connected-react-router#429]
Hi,

It seems that history v5 has somme breaking changes, and conected-react-router is not compatible with this version.
See issue posted on history repository: [https://github.com/ReactTraining/history/issues/803](https://github.com/ReactTraining/history/issues/803)

**Comment by stage88** [comment]
I found i am getting a `Could not find router reducer in state tree, it must be mounted under "router"`, i traced it down to the missing `action` prop on the persisted route object, as @readikus mentioned above.

in selectors.js, there is code looking for it when validating if it is a router:
```
var isRouter = function isRouter(value) {
  return value != null && _typeof(value) === 'object' && getIn(value, ['location']) && getIn(value, ['action']);
};
```

**Comment by codingwithmanny** [comment]
Solution for now, found here;
https://github.com/ReactTraining/history/issues/804#issuecomment-646452641

**TL;DR:**
```bash
yarn add history@4.10.1
```

**Comment by enterthevoid** [comment]
> Solution for now, found here;
> [ReactTraining/history#804 (comment)](https://github.com/ReactTraining/history/issues/804#issuecomment-646452641)
> 
> **TL;DR:**
> 
> ```shell
> yarn add history@4.10.1
> ```

Works for me

**Comment by nandenjin** [comment]
Is there any roadmap to fix this problem (without downgrade to `4.x`)?

**Comment by supasate** [comment]
This is due to React Router v5 doesn't support history v5. 

When adding history v5, it should show a warning that it has incorrect peer dependency.

React Router v6 will support history v5. So, we will upgrade it as well when it's out of beta.

**Comment by chris-tuncap** [comment]
> This is due to React Router v5 doesn't support history v5.
> 
> When adding history v5, it should show a warning that it has incorrect peer dependency.
> 
> React Router v6 will support history v5. So, we will upgrade it as well when it's out of beta.

React Router v6 is out now and supports history v5.  Can you upgrade it now?

**Comment by dkadrios** [comment]
@chris-tuncap, my team switched over to redux-first-history https://www.npmjs.com/package/redux-first-history

Works great with v6 and react 18
